packages/filebrowser-extension/src/index.ts

```diff
@@ -261,6 +274,26 @@ function activateBrowser(
     shell.layoutModified.connect(() => {
       maybeCreate();
     });
+
+    // Whether to automatically navigate to a document's current directory
+    shell.currentChanged.connect((shell, change) => {
+      settingRegistry
```

Comment:
I don't think we need to load the settings from the server on every `currentChanged`. We should be able to do it once, store `navigateToCurrentDirectory`, and keep it up to date on `settingsChanged`.